src/pageCreation/createLatestLegalNotices.js

```diff
@@ -0,0 +1,54 @@
+const { LEGAL_NOTICES_QUERY } = require('../queries/gatsbyNodeQueries');
+const path = require(`path`);
+const has = require('lodash/has');
+const map = require('lodash/map');
+const get = require('lodash/get');
+
+/*
+ * Legal notices are immutable and versioned so they show up under different routes such as
+ * \/legal-noices\/sub-processors\/v1\/. This is good because people can go back in time and
+ * look at different versions of the legal notices we have used.
+ *
+ * At the same time, it is useful to quickly be able to see what the latest version of a
+ * legal notice is. This is the version we want to link to in the sidebar of the legal notices
+ * page.
+ *
+ * This function finds all different types of legal notices, then figures out the latest one
+ * based on the version number, then makes that the "default" one.
+ */
+const createLatestLegalNotices = async ({ graphql, actions: { createPage } }) => {
+  const component = path.resolve(__dirname, '../templates/LegalNotice.js');
+  const { data, errors } = await graphql(LEGAL_NOTICES_QUERY);
+
+  if (errors) {
+    throw errors;
+  }
+
+  const maxVersionForType = get(data, 'notices.edges').reduce((accumulator, edge) => {
```

Comment:
An alternative implementation that could look simpler to read is to map them first to each type and then sort them?

Reply:
That version looks like this. What do you think?

```js
  const versionsForEachType = get(data, 'notices.edges').reduce((accumulator, edge) => {
    const matches = edge.node.fields.slug.match(/\/legal-notices\/([\w-]+)\/v(\d+)\//);
    const noticeType = matches[1];
    const version = parseInt(matches[2]);

    if (has(accumulator, noticeType)) {
      accumulator[noticeType] = [...accumulator[noticeType], version];
    } else {
      accumulator[noticeType] = [version];
    }

    return accumulator;
  }, {});

  const maxVersionForType = reduce(versionsForEachType, (accumulator, versions, noticeType) => {
    accumulator[noticeType] = max(versions);
    return accumulator;
  }, {});
```

Reply:
hm, it doesn't really look much simpler so up to you :shrug: no strong opinion here